contracts/src/v0.8/workflow/test/WorkflowRegistryManager/WorkflowRegistryManager.addVersion.t.sol

```diff
@@ -1,32 +1,103 @@
-    _;
+  // whenTheCallerIsTheOwner
+  function test_RevertWhen_TheContractAddressIsInvalid() external {
+    // Deploy a random contract
+    MockContract mockContract = new MockContract();
+
+    // Add it to the WorkflowRegistryManager
+    vm.prank(s_owner);
+    vm.expectRevert(abi.encodeWithSelector(WorkflowRegistryManager.InvalidContractType.selector, address(mockContract)));
+    s_registryManager.addVersion(address(mockContract), s_chainID, s_deployedAt, true);
   }
 
-  function test_RevertWhen_TheContractAddressIsInvalid() external whenTheCallerIsTheOwner {
-    // it should revert
+  // whenTheCallerIsTheOwner
+  function test_RevertWhen_TheContractAddressIsValid() external {
+    // Add a 0 address to the WorkflowRegistryManager
+    vm.prank(s_owner);
+    vm.expectRevert(abi.encodeWithSelector(WorkflowRegistryManager.InvalidContractAddress.selector, address(0)));
+    s_registryManager.addVersion(address(0), s_chainID, s_deployedAt, true);
   }
 
-  modifier whenTheContractAddressIsValid() {
-    _;
+  // whenTheCallerIsTheOwner whenTheContractAddressIsValid
+  function test_RevertWhen_TheContractTypeIsInvalid() external {
+    // Deploy a random contract
+    MockContract mockContract = new MockContract();
+
+    // Add it to the WorkflowRegistryManager
+    vm.prank(s_owner);
+    vm.expectRevert(abi.encodeWithSelector(WorkflowRegistryManager.InvalidContractType.selector, address(mockContract)));
+    s_registryManager.addVersion(address(mockContract), s_chainID, s_deployedAt, true);
   }
 
-  function test_WhenAutoActivateIsTrue() external whenTheCallerIsTheOwner whenTheContractAddressIsValid {
-    // it should deactivate any currently active version
-    // it should activate the new version
-    // it should emit VersionAddedV1 after adding the version to s_versions
-    // it should emit VersionActivatedV1
+  // whenTheCallerIsTheOwner whenTheContractAddressIsValid whenTheContractTypeIsValid
+  function test_WhenAutoActivateIsTrue() external {
+    // Get the latest version number, which should revert.
+    vm.expectRevert(WorkflowRegistryManager.NoVersionsRegistered.selector);
+    s_registryManager.getLatestVersionNumber();
+
+    // Deploy a MockWorkflowRegistryContract contract
+    MockWorkflowRegistryContract mockWfrContract = new MockWorkflowRegistryContract();
+
+    // Expect both VersionAdded and VersionActivated events to be emitted.
+    vm.expectEmit(true, true, false, true);
+    emit WorkflowRegistryManager.VersionAdded(address(mockWfrContract), s_chainID, s_deployedAt, 1);
+    emit WorkflowRegistryManager.VersionActivated(address(mockWfrContract), s_chainID, 1);
+
+    // Add the MockWorkflowRegistryContract to the WorkflowRegistryManager
+    vm.prank(s_owner);
+    s_registryManager.addVersion(address(mockWfrContract), s_chainID, s_deployedAt, true);
+
+    // Get the latest version number again, which should be 1.
+    uint32 versionNumber = s_registryManager.getLatestVersionNumber();
```

Comment:
Nit: maybe there would be more differences between this test and `test_WhenAutoActivateIsFalse` if another active version already exists on the contract. That means this has to return 2, and for the other test, it has to return 1.